### PR TITLE
[WIP] Migrate terraform-aws-ecs-scheduled-task to Terraform 0.12

### PR DESCRIPTION
**Pull request by JonRoma**

### CODE REVIEW

**Comment by JonRoma** [comment]
@ddriddle, I think we ought to have an example to test this code, and I think that is the reason I did not create the PR for your review. It passes all the Drone tests, FWIW, but we need to actually run it first!

File: data.tf [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
 
-  tags {
-    Tier = "${local.tier}"
+  tags = {
+    Tier = local.tier
   }
 }
 
-data "aws_caller_identity" "current" {}
+data "aws_caller_identity" "current" {
```
Comment (ddriddle) [inline data.tf]: Join.

File: locals.tf [hunk 2]
```diff
@@ -1,29 +1,41 @@
 # task_defintion_arn
 locals {
-  task_definition_arn = "${element(concat(aws_ecs_task_definition.fargate.*.arn,
-                           aws_ecs_task_definition.ec2.*.arn,  list("")), 0)}"
+  task_definition_arn = element(
+    concat(
```
Comment (ddriddle) [inline locals.tf]: This could be partially joined.

File: locals.tf [hunk 3]
```diff
@@ -1,29 +1,41 @@
...
 }
 
 # task_definition map
 locals {
-  container_definition_file = "${lookup(var.task_definition, "container_definition_file", "containers.json")}"
-  cpu                       = "${lookup(var.task_definition, "cpu", "256")}"
-  memory                    = "${lookup(var.task_definition, "memory", "512")}"
-  network_mode              = "${lookup(var.task_definition, "network_mode", "awsvpc")}"
-  task_role_arn             = "${lookup(var.task_definition, "task_role_arn", "")}"
+  container_definition_file = lookup(
```
Comment (ddriddle) [inline locals.tf]: Join.

File: main.tf [hunk 4]
```diff
@@ -1,61 +1,69 @@
...
+  count = var.task_definition_arn == "" && var.launch_type == "FARGATE" ? 1 : 0
 
-  family                = "${var.name}"
-  container_definitions = "${file(local.container_definition_file)}"
-  task_role_arn         = "${local.task_role_arn}"
-  execution_role_arn    = "${format("arn:aws:iam::%s:role/ecsTaskExecutionRole", data.aws_caller_identity.current.account_id)}"
+  family                = var.name
+  container_definitions = file(local.container_definition_file)
+  task_role_arn         = local.task_role_arn
+  execution_role_arn = format(
```
Comment (ddriddle) [inline main.tf]: Consider joining.

**Pull request merged**